### PR TITLE
Update from_app.rst

### PR DESCRIPTION
**Pull request by karvell**
Consolidated and clarified this topic so users don't have to parse partial MSDN documentation that may or may not apply for the wopi integration. 
I'm not sure how to do the numbering under Identifying supported Office versions. Please read to make sure all the formatting, etc., looks correct.

Thanks!

### CODE REVIEW

File: docs/native/src/scenarios/from_app.rst [hunk 1]
```diff
@@ -9,42 +9,45 @@ Opening files from your app in |Office iOS|
 In order to invoke |Office iOS| when opening an Office file from your app, you must use the URL schemes that Word,
 Excel, and PowerPoint for iOS register when they are installed.
 
-General information on the Office URL schemes can be found here:
-https://msdn.microsoft.com/en-us/library/office/dn906146.aspx#sectionSection1
+URL Schemes for Inovking Office
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Typo. If you set up a local build environment you can actually run check_spelling.bat to check your spelling. :smile:

File: docs/native/src/scenarios/from_app.rst [hunk 2]
```diff
@@ -9,42 +9,45 @@ Opening files from your app in |Office iOS|
...
+-------------------------------
 
-In order to open files on your WOPI host, you must do the following:
+The following is the list of scheme names used to invoke Office:
 
-* Pass the :term:`WOPISrc` to the file via the URL schemes
-* Along with the WOPISrc of the file, you must include:
+ * ms-word:
+ * ms-powerpoint:
+ * ms-excel:
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Remove the leading spaces.

File: docs/native/src/scenarios/from_app.rst [hunk 3]
```diff
@@ -9,42 +9,45 @@ Opening files from your app in |Office iOS|
...
 
-  * The service identifier, denoted by the ``|d|`` parameter
-  * The user identifier, denoted by the ``|e|`` parameter
-  * The file name, with extension, denoted by the ``|n|`` parameter
+The following informaiton must be included along with the URL scheme:
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Typo.

File: docs/native/src/scenarios/from_app.rst [hunk 4]
```diff
@@ -9,42 +9,45 @@ Opening files from your app in |Office iOS|
...
+ * ms-word:
+ * ms-powerpoint:
+ * ms-excel:
 
-  * The service identifier, denoted by the ``|d|`` parameter
-  * The user identifier, denoted by the ``|e|`` parameter
-  * The file name, with extension, denoted by the ``|n|`` parameter
+The following informaiton must be included along with the URL scheme: 
 
+* The mode to open the file in. Values values are:
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Typo - I think you mean "valid values."

File: docs/native/src/scenarios/from_app.rst [hunk 5]
```diff
@@ -9,42 +9,45 @@ Opening files from your app in |Office iOS|
...
+ * ``ofv`` for opening as read-only
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Add two spaces for nested lists. The `*` should align with the text of the parent bullet.

File: docs/native/src/scenarios/from_app.rst [hunk 6]
```diff
@@ -9,42 +9,45 @@ Opening files from your app in |Office iOS|
...
-  * ``Web`` – to be used in the case where a website is invoking the protocol handler
-  * ``App`` – to be used in the case where a native app is invoking the protocol handler
+ * ``Web`` – to be used in the case where a website is invoking the protocol handler
+ * ``App`` – to be used in the case where a native app is invoking the protocol handler
 
 Example::
 
     ms-word:ofe|u|https://contoso/wopi/file/12312|d|Contoso|e|a3243d|n|document1.docx|a|App
 
+The URL used should be encoded.
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Encoded how?

File: docs/native/src/scenarios/from_app.rst [hunk 7]
```diff
@@ -9,42 +9,45 @@ Opening files from your app in |Office iOS|
...
-In addition to the parameters described above, you may also include the passback parameters described in
-https://msdn.microsoft.com/EN-US/library/office/dn911482.aspx. This enables an in-app back button in Office that
-takes the user back to your app. This is in addition to the iOS9 platform-provided back button.
-
-
 Identifying supported Office versions
 -------------------------------------
 
-Before using the above URL schemes to invoke Office, you must do the following:
+Before using the URL schemes specificed above to invoke Office, you must do the following:
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Typo.

File: docs/native/src/scenarios/from_app.rst [hunk 8]
```diff
@@ -9,42 +9,45 @@ Opening files from your app in |Office iOS|
...
 
-#.  Check that Office is installed. This can be done by verifying whether the Office URL schemes are registered. See
-    https://msdn.microsoft.com/EN-US/library/office/dn911482.aspx for more information.
+#.  Verify the particular Office appicaiton (Word, Excel, or PowerPoint) are installed.
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Typo

File: docs/native/src/scenarios/from_app.rst [hunk 9]
```diff
@@ -56,5 +59,25 @@ Before using the above URL schemes to invoke Office, you must do the following:
...
         These URL schemes are only used to check if the currently installed Office supports opening files from a WOPI
-        host and not for invoking the Office apps. Invoking the apps is done using the URL schemes described at
-        https://msdn.microsoft.com/en-us/library/office/dn906146.aspx#sectionSection3.
+        host and not for invoking the Office apps. 
+
+(Optional) Passback protocol
+----------------------------
+If you want Office to return users to your iOS application when they choose the in app back arrow (distinct from the iOS back control), you will need to inlcude the passback parameter when invoking Office. This is dnoted by ``|p|`` followed by your app's registerd URL scheme (without a colon). 
+
+You’ll need to ensure that your application can properly handle the response from Office.
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Avoid the 'smart quotes' if you can. Plain text `'` instead.

File: docs/native/src/scenarios/from_app.rst [hunk 10]
```diff
@@ -56,5 +59,25 @@ Before using the above URL schemes to invoke Office, you must do the following:
     ..  important::
 
         These URL schemes are only used to check if the currently installed Office supports opening files from a WOPI
-        host and not for invoking the Office apps. Invoking the apps is done using the URL schemes described at
-        https://msdn.microsoft.com/en-us/library/office/dn906146.aspx#sectionSection3.
+        host and not for invoking the Office apps. 
+
+(Optional) Passback protocol
+----------------------------
+If you want Office to return users to your iOS application when they choose the in app back arrow (distinct from the iOS back control), you will need to inlcude the passback parameter when invoking Office. This is dnoted by ``|p|`` followed by your app's registerd URL scheme (without a colon).
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Typo.

File: docs/native/src/scenarios/from_app.rst [hunk 11]
```diff
@@ -56,5 +59,25 @@ Before using the above URL schemes to invoke Office, you must do the following:
...
-        https://msdn.microsoft.com/en-us/library/office/dn906146.aspx#sectionSection3.
+        host and not for invoking the Office apps. 
+
+(Optional) Passback protocol
+----------------------------
+If you want Office to return users to your iOS application when they choose the in app back arrow (distinct from the iOS back control), you will need to inlcude the passback parameter when invoking Office. This is dnoted by ``|p|`` followed by your app's registerd URL scheme (without a colon). 
+
+You’ll need to ensure that your application can properly handle the response from Office.
+
+Plesae provide your app's registered URL scheme as part of onboarding with Microsoft.
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Typo.

File: docs/native/src/scenarios/from_app.rst [hunk 12]
```diff
@@ -56,5 +59,25 @@ Before using the above URL schemes to invoke Office, you must do the following:
...
+
+(Optional) Passback protocol
+----------------------------
+If you want Office to return users to your iOS application when they choose the in app back arrow (distinct from the iOS back control), you will need to inlcude the passback parameter when invoking Office. This is dnoted by ``|p|`` followed by your app's registerd URL scheme (without a colon). 
+
+You’ll need to ensure that your application can properly handle the response from Office.
+
+Plesae provide your app's registered URL scheme as part of onboarding with Microsoft. 
+
+For security reasons, Office only returns users to the referring application if the file opened successfully. When the user chooses the back arrow, Office responds to the invoking application with the passback protocol, open mode, URL, upload pending status, and ``document context``. The upload pending status uses the descriptor |z|, and is either yes or no.
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Put the `|z|` in backticks too, so it gets 'code' styling applied to it. I.e.

```
``|z|``
```

File: docs/native/src/scenarios/from_app.rst [hunk 13]
```diff
@@ -56,5 +59,25 @@ Before using the above URL schemes to invoke Office, you must do the following:
...
+
+You’ll need to ensure that your application can properly handle the response from Office.
+
+Plesae provide your app's registered URL scheme as part of onboarding with Microsoft. 
+
+For security reasons, Office only returns users to the referring application if the file opened successfully. When the user chooses the back arrow, Office responds to the invoking application with the passback protocol, open mode, URL, upload pending status, and ``document context``. The upload pending status uses the descriptor |z|, and is either yes or no. 
+
+``document context`` is a string you provide via the ``|c|`` parameter when invoking Office. Office doesn't use this parameter; it is purely for your use, as needed by your app.  Office does not limit the length of the string, beyond any limits imposed by the operating system. 
+
+Schema format inovking your app when user choses back::
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Typo

File: docs/native/src/scenarios/from_app.rst [hunk 14]
```diff
@@ -56,5 +59,25 @@ Before using the above URL schemes to invoke Office, you must do the following:
...
+
+    <app protocol>:ofe|u|<URL>|z|<yes or no>|c|<doc context> 
+
+Example::
+
+    clouddrive:ofe|u|https://contoso/Q4/budget.docx|z|no|c|folderviewQ4
```
Comment (tylerbutler) [inline docs/native/src/scenarios/from_app.rst]: Maybe `contosodrive` instead of `clouddrive`?

File: docs/native/src/scenarios/from_app.rst (hunk 1 above)
Comment (karvell) [inline docs/native/src/scenarios/from_app.rst]: I believe I fixed all the issues you provided feedback on.
But I’m not sure how to do the check_spelling.bat though. I have the GitHub app which really only helps me manage the branches, not with actual editing. check_spelling doesn’t seem to work in powershell.

Can you provide some pointers on how to do this?

Thanks,
Karvell

**Comment by tylerbutler** [comment]
A squashed version of this was actually merged in in commit a36a5d7aec526985849d2c487100b53b0dd1e8a0.

**Pull request closed without merging**